crates/proof-of-sql/src/sql/postprocessing/group_by_expr.rs

```diff
@@ -0,0 +1,245 @@
+        let (agg_exprs, remainder) = aliased_exprs
+            .iter()
+            .map(|aliased_expr| -> PostprocessingResult<_> {
+                check_and_get_aggregation_and_remainder(aliased_expr, by_ids, &mut agg_count)
+            })
+            .collect::<PostprocessingResult<Vec<_>>>()?;
+        Ok(Self {
+            result_exprs: remainder,
+            group_by_identifiers: group_by_identifier_set.into_iter().collect::<Vec<_>>(),
+            agg_expr_pairs: agg_exprs.into_iter().flatten().collect(),
+        })
+    }
+}
+
+impl<S: Scalar> PostprocessingStep<S> for GroupByExpr {
+    /// Apply the group by transformation to the given `OwnedTable`.
+    fn apply(&self, owned_table: OwnedTable<S>) -> PostprocessingResult<OwnedTable<S>> {
+        // First evaluate all the aggregated columns
+        let evaluated_columns: Vec<(AggregationOperator, OwnedColumn<S>, Identifier)> = self
+            .agg_expr_pairs
+            .iter()
+            .map(|(agg_op, expr, id)| {
+                let evaluated_column = expr.evaluate(&owned_table)?;
+                Ok((agg_op, evaluated_column, id))
+            })
+            .collect::<PostprocessingResult<Vec<_>>>()?;
+        // Next actually do the GROUP BY
+        let alloc = Bump::new();
+        let group_by_columns = self
+            .group_by_identifiers
+            .iter()
+            .map(|id| {
+                let column = owned_table
+                    .inner_table()
+                    .get(id)
+                    .ok_or(PostprocessingError::ColumnNotFound(id.to_string()))?;
+                Ok(Column::<S>::from_owned_column(column, &alloc))
+            })
+            .collect::<PostprocessingResult<Vec<_>>>()?;
+        let (sum_columns, sum_ids) = evaluated_columns
+            .iter()
+            .filter(|(agg_op, _owned_col, _id)| agg_op == AggregationOperator::Sum)
+            .map(|(agg_op, owned_col, id)| (Column::<S>::from_owned_column(owned_col, &alloc), id))
+            .unzip();
+        let (max_columns, max_ids) = evaluated_columns
+            .iter()
+            .filter(|(agg_op, _owned_col, _id)| agg_op == AggregationOperator::Max)
+            .map(|(agg_op, owned_col, id)| (Column::<S>::from_owned_column(owned_col, &alloc), id))
+            .collect::<Vec<_>>();
+        let (min_columns, min_ids) = evaluated_columns
+            .iter()
+            .filter(|(agg_op, _owned_col, _id)| agg_op == AggregationOperator::Min)
+            .map(|(agg_op, owned_col, id)| (Column::<S>::from_owned_column(owned_col, &alloc), id))
+            .collect::<Vec<_>>();
+        let (count_columns, count_ids) = evaluated_columns
+            .iter()
+            .filter(|(agg_op, _owned_col, _id)| agg_op == AggregationOperator::Count)
+            .map(|(agg_op, owned_col, id)| (Column::<S>::from_owned_column(owned_col, &alloc), id))
+            .collect::<Vec<_>>();
+        // TODO: Allow a filter
```

Comment:
Is this TODO needed for postprocessing?

Reply:
@JayWhite2357 Not yet. It is quite trivial to accept non-provable filters though.